source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,372 @@
+``src/Controller`` directory and paste the following code to handle the
+endpoints in your application:
+
+.. literalinclude:: /includes/php-frameworks/symfony/RestaurantController.php
+   :caption: src/Controller/RestaurantController.php
+   :language: php
+
+The controller file defines the ``index()`` method, which displays text on
+the web app's front page. The file also defines the ``browse()`` method,
+which finds documents in which the ``borough``
+field is ``'Queens'`` and the ``name`` field contains the string ``'Moon'``.
+This method then displays the documents at the ``/restaurant/browse/`` route. The
+``browse()`` method uses the ``QueryBuilder`` class to construct the query.
+
+Specify Routes
+``````````````
+
+In the ``config/routes.yaml`` file, add the following routes:
+
+.. code-block:: yaml
+   :caption: config/routes.yaml
+
+   # main routes
+   restaurant_index:
+     path: /
+     controller: App\Controller\RestaurantController::index
+     methods: [GET]
+   
+   # browse restaurants
+   restaurant_browse:
+     path: /restaurant/browse
+     controller: App\Controller\RestaurantController::browse
+     methods: [GET]
+
+This file sets the following routes in the application:
+
+- ``restaurant_index [GET]``: landing page
+- ``restaurant_browse [GET]``: retrieves and displays specific restaurants
+
+Customize Templates
+```````````````````
+
+Next, create templates to customize the web app's appearance.
+
+Create the ``templates/restaurant`` directory and populate it with the
+following files:
+
+- ``index.html.twig``
+- ``browse.html.twig``
+
+Paste the following code into the ``index.html.twig`` file:
+
+.. literalinclude:: /includes/php-frameworks/symfony/index.html.twig
+   :caption: templates/restaurant/index.html.twig
+   :language: html
+
+Paste the following code into the ``browse.html.twig`` file:
+
+.. literalinclude:: /includes/php-frameworks/symfony/browse.html.twig
+   :caption: templates/restaurant/browse.html.twig
+   :language: html
+
+Start your Symfony Application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run the following command from the application root directory to start
+your PHP built-in web server:
+
+.. code-block:: bash
+   
+   symfony server:start
+
+After the server starts, it outputs the following message:
+
+.. code-block:: none
+   :copyable: false
+
+   [OK] Web server listening
+        The Web server is using PHP FPM 8.3.4
+        http://127.0.0.1:8000
+
+Open the URL http://127.0.0.1:8000/restaurant/browse in your web browser.
+The page shows a list of restaurants and details about each of them, as
+displayed in the following screenshot:
+
+.. figure:: /includes/php-frameworks/symfony/app_render.png
+   :alt: Restaurants web app screenshot
+
+Congratulations on completing the Quick Start tutorial!
+
+After you complete these steps, you have a Symfony web application that
+connects to your MongoDB deployment, runs a query on
+the sample data, and renders a retrieved result.
```

Comment:
If you care, the line above this wraps prematurely.

Reply:
SHould not make a difference for staging but good eye!